tests/ui/error-codes/E0001.stderr

```diff
@@ -2,8 +2,17 @@ error: unreachable pattern
   --> $DIR/E0001.rs:8:9
    |
 LL |         _ => {/* ... */}
-   |         ^
+   |         ^ unreachable pattern
    |
+note: these patterns collectively make the last one unreachable
+  --> $DIR/E0001.rs:8:9
+   |
+LL |         Some(_) => {/* ... */}
+   |         ------- matches some of the same values
+LL |         None => {/* ... */}
+   |         ---- matches some of the same values
+LL |         _ => {/* ... */}
+   |         ^ collectively making this unreachable
```

Comment:
I find the flat line a bit hard to parse, looks more like a separator or a leader to another arrow than an indicator itself. Any specific reason this uses the spatula rather than the forks?

(regardless, this change is wonderful)

Reply:
I don't know what "spatula" and "forks" mean, I think just used the default principal/non-principal span mechanism in diagnostics

Reply:
I was being too creative, but

`______--` spatula

`^^---` fork

I figured it wasn't anything done specifically here. This diagnostic just winds up with more alternating lines of code and `----` than many others, and I think it is somewhat ambiguous whether `----` refers to the line up or down.

Obviously not a big deal since it is just a hint anyway.

Reply:
I see. It looks a lot clearer with color at least